ui/selectors/selectors.js

```diff
@@ -1823,12 +1823,35 @@ export function getNumberOfAllUnapprovedTransactionsAndMessages(state) {
 export const getCurrentNetwork = createDeepEqualSelector(
   getAllNetworks,
   getProviderConfig,
+  /**
+   * Get the current network configuration.
+   *
+   * @param {Record<string, unknown>[]} allNetworks - All network configurations.
+   * @param {Record<string, unknown>} providerConfig - The configuration for the current network's provider.
+   * @returns {{
+   *   chainId: `0x${string}`;
+   *   id?: string;
+   *   nickname?: string;
+   *   providerType?: string;
+   *   rpcPrefs?: { blockExplorerUrl?: string; imageUrl?: string; };
+   *   rpcUrl: string;
+   *   ticker: string;
+   * }} networkConfiguration - Configuration for the current network.
+   */
   (allNetworks, providerConfig) => {
     const filter =
       providerConfig.type === 'rpc'
         ? (network) => network.id === providerConfig.id
         : (network) => network.id === providerConfig.type;
-    return allNetworks.find(filter);
+    return (
+      allNetworks.find(filter) ?? {
+        chainId: providerConfig.chainId,
```

Comment:
The `providerConfig.type` property is being excluded here because that's not an expected network configuration property.

Reply:
Have we removed instances like the one causing [this error](https://metamask.sentry.io/issues/5660992353/?environment=production&project=273505&query=firstRelease%3A12.0.0&referrer=issue-stream&sort=freq&statsPeriod=90d&stream_index=12) expecting an `id`?

Reply:
Looks like it's still in this file down on line 1912?

Reply:
To clarify, the `id` property is still expected to be present in cases where `providerConfig` matches a valid custom network configuration. It would cause problems if anyplace used this selector and expected an `id` to be _always_ present, but the line you linked works fine if `id` is `undefined`. If the current domain has a selected network configuration set, it'll find that it doesn't match `undefined` and it will switch, which is correct.

Reply:
ah ok right. The issue I was thinking of was when `currentNetwork` is undefined not `id`: https://metamask.sentry.io/issues/5660992353/?environment=production&project=273505&query=firstRelease%3A12.0.0&referrer=issue-stream&sort=freq&statsPeriod=90d&stream_index=12